code/modules/spells/spell_types/vampire_powers.dm

```diff
@@ -0,0 +1,416 @@
+
+		if(blood_used <= vampire.usable_blood)
+			vampire.usable_blood -= blood_used
+		else
+			// stop!!
+			targets.Cut()
+
+		if(LAZYLEN(targets))
+			to_chat(usr, "<span class='notice'><b>You have [vampire.usable_blood] left to use.</b></span>")
+
+
+/obj/effect/proc_holder/spell/can_target(mob/living/target)
+	. = ..()
+	if(vamp_req && is_vampire(target))
+		return FALSE
+/datum/vampire_passive
+	var/gain_desc
+
+/datum/vampire_passive/New()
+	..()
+	if(!gain_desc)
+		gain_desc = "You have gained \the [src] ability."
+
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/datum/vampire_passive/regen
+	gain_desc = "Your rejuvination abilities have improved and will now heal you over time when used."
+
+/datum/vampire_passive/vision
+	gain_desc = "Your vampiric vision has improved."
+
+/datum/vampire_passive/full
+	gain_desc = "You have reached your full potential and are no longer weak to the effects of anything holy and your vision has been improved greatly."
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/effect/proc_holder/spell/self/rejuvenate
+	name = "Rejuvenate"
+	desc= "Flush your system with spare blood to remove any incapacitating effects."
+	action_icon_state = "rejuv"
+	charge_max = 200
+	stat_allowed = 1
+	action_icon = 'icons/mob/vampire.dmi'
+	action_background_icon_state = "bg_demon"
+	vamp_req = TRUE
+
+/obj/effect/proc_holder/spell/self/rejuvenate/cast(list/targets, mob/user = usr)
+	var/mob/living/carbon/U = user
+	U.stuttering = 0
+	U.SetAllImmobility(0)
+	U.set_resting(FALSE)
+	U.adjustStaminaLoss(-75)  //Hey, lets not just stand in one place because we are too tired to move
+
+	var/datum/antagonist/vampire/V = U.mind.has_antag_datum(/datum/antagonist/vampire)
+	if(!V) //sanity check
+		return
+	for(var/i = 1 to 5)
+		U.adjustStaminaLoss(-20)
+		if(V.get_ability(/datum/vampire_passive/regen))
+			U.adjustBruteLoss(-2)
+			U.adjustOxyLoss(-5)
+			U.adjustToxLoss(-2, forced = TRUE)
+			U.adjustFireLoss(-2)
+		sleep(7.5)
+
+
+/obj/effect/proc_holder/spell/targeted/hypnotise
+	name = "Hypnotize (30)"
+	desc= "A piercing stare that knocks out your victim for a short lenght of time"
+	blood_used = 30
+	charge_max = 500  //No, you cant chain this with multiple people.
+	action_icon_state = "hypnotize"
+	action_icon = 'icons/mob/vampire.dmi'
+	action_background_icon_state = "bg_demon"
+	vamp_req = TRUE
+
+//Todo, make the targeted abilities auto activate if theres only one target in range.
+/obj/effect/proc_holder/spell/targeted/hypnotise/cast(list/targets, mob/user = usr)
```

Comment:
So only counter to this is "If anti magic"
Imo should check blind/holy traits as well...
Maybe even if flash 3+ aka blind but do to layers of protection/damage?
Just something other then chaplin weapon tehe

Reply:
i'll probably add blind but only blind.